lib/nylas/message.rb

```diff
@@ -46,6 +50,15 @@ def unread?
       unread
     end
 
+    def update(data)
+      unupdatable_attributes = data.keys.reject { |name| UPDATABLE_ATTRIBUTES.include?(name) }
```

Comment:
What do you think about extracting a private method, something named like `def unupdatable_attributes?` to raise the exception. Then it would maybe look like

```ruby
def update(data)
  if unupdatable_attributes?(data)
    raise ArgumentError, # message
  else  
    super(data)
  end
end
```

?

Reply:
Make sense.

Reply:
I have added a new class to filter attributes here https://github.com/nylas/nylas-ruby/pull/231/commits/e40e7cee254ced5f4b22d57b7932a65694fea700

I think we are going to need this class everywhere where we don't want to send unwanted params to server.

Reply:
Oh this is beautiful ✨